### ISSUE TITLE
Explicitly support renewal of loans

### ISSUE THREAD

**Issue by leonardr** [NYPL-Simplified/circulation#205]
Currently our API supports renewal sort of by accident. If you hit the `borrow` link and your book can be renewed, it gets renewed.

The problem is that you might not know you were renewing the book in the first place. If your book can't be renewed, and you didn't know you were renewing, the error conditions you get can be confusing.

In addition, there is no signal in the OPDS feed that a book can in fact be renewed, meaning that it only ever gets renewed by accident.

The solution is to define a new link relation for renewal (or take one from ODL) and present it when appropriate. The link should point to a distinct API endpoint which returns different error codes than the normal `borrow` endpoint would in the exact same scenarios.

For example, if you already have a loan for a book, and you try to borrow the book again, the Overdrive API may raise `NoLicensesAvailable`. This exception should be ignored and you should just get a reference to your preexisting loan. 

If you already have a loan for a book and you try to renew the loan, the Overdrive API may raise `NoLicensesAvailable`. It's the exact same case, but in this case the exception should _not_ be ignored. It's telling you that you can't renew the book because other patrons have it on hold.

**Comment by HadrienGardeur** [comment]
I think there's a fairly important question to ask first: is renewal a feature of the DRM license or a feature of the OPDS catalog?

**Comment by leonardr** [comment]
The fact that you can't renew a license if there are people waiting in line indicates it's a property of the catalog. Whether or not to grant the renewal depends on library policy. The way it's implemented in Overdrive, the patron is effectively asking to create a new loan back-to-back on their existing loan. The output of this process is an updated ACSM file.

In theory, renewal could be an operation handled entirely by the DRM system, but I think that would only work in an environment where there's no limit on how many people can access the book at once.

**Comment by HadrienGardeur** [comment]
We might want to separate renewing from extending a loan then.

Agree it's part of the library policy, but it's also tied to the DRM.

While ACS doesn't stop you from renewing if you consider a new loan to be a renewal, it doesn't support extensions at all.

**Comment by leonardr** [comment]
I agree that creating a new back-to-back loan is different from extending the expiration period on an existing loan, and I'm OK with using distinct link relations for the two operations. However I don't think a patron really cares about the difference, so if a site supports both operations (as Simplified probably will since it integrates different providers) I'm not sure what that should look like in the UI. The worst would be if a given book supported both operations, though that seems a little far-fetched.

**Comment by HadrienGardeur** [comment]
I agree that from a patron perspective things should be transparent.

But from a library perspective, this could be very different with a separate policy if they want to.

From an ODL perspective, every ebook already supports renewal since in ODL the content provider is not in charge of the hold list.
Extending a loan will also be supported, but strictly in LCP for now and will only show up in the status document for a loan (won't show up while harvesting metadata from OPDS or when you request the copy status document).
